packages/pyright-internal/src/tests/languageServer.test.ts

```diff
@@ -235,4 +236,301 @@ describe(`Basic language server tests`, () => {
+                    CancellationToken.None
+                );
+                assertEqual(result, {
+                    documentChanges: [
+                        {
+                            edits: [
+                                {
+                                    range: {
+                                        start: {
+                                            line: 0,
+                                            character: 7,
+                                        },
+                                        end: {
+                                            line: 0,
+                                            character: 10,
+                                        },
+                                    },
+                                    newText: 'foo2',
+                                },
+                                {
+                                    range: {
+                                        start: {
+                                            line: 1,
+                                            character: 0,
+                                        },
+                                        end: {
+                                            line: 1,
+                                            character: 3,
+                                        },
+                                    },
+                                    newText: 'foo2',
+                                },
+                            ],
+                            textDocument: {
+                                uri: 'file:///src/baz.py',
+                                version: null,
+                            },
+                        },
+                        {
+                            edits: [],
+                            textDocument: {
+                                uri: marker.fileUri.toString(),
+                                version: null,
+                            },
+                        },
+                    ],
+                });
+            });
+
+            test('rename __init__ file to module (currently not supported so no edits should be created)', async () => {
+                const code = `
+// @filename: foo/__init__.py
+//// # empty file [|/*marker*/|]
+//// 
+// @filename: baz.py
+//// import foo
+//// foo
+//// 
+    `;
+                const serverInfo = await runLanguageServer(DEFAULT_WORKSPACE_ROOT, code, true);
+                openFile(serverInfo, 'marker');
+                const marker = serverInfo.testData.markerPositions.get('marker')!;
+                const result = await serverInfo.connection.sendRequest(
+                    WillRenameFilesRequest.type,
+                    {
+                        files: [{ oldUri: marker.fileUri.toString(), newUri: 'file:///src/foo/bar.py' }],
+                    },
+                    CancellationToken.None
+                );
+                assertEqual(result, {
+                    documentChanges: [],
```

Comment:
does this show an error to the user? silently not work, or silently break? and if there is a message, should there be a test for it?

Reply:
no message, it just silently doesn't do it, same as pylance.

since this functionality is implemented at the language server level i don't know if there's a way to show a message to the user unless i make it vscode specific. i will just raise a separate issue to support it

Reply:
Resolve once raised

Reply:
#498